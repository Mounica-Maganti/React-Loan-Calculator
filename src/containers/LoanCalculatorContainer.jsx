import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import LoanCalculator from '../components/LoanCalculator/LoanCalculator';
import LoanCalculatorSummary from '../components/LoanCalculatorSummary/LoanCalculatorSummary';
import * as mapDispatchToProps from '../actions/index';
import '../custom.css';

class LoanCalculatorContainer extends Component {

  render() {
    const { loanCalculator: { loanInfo, showLoanCalculator, showLoanSummary, summaryData }, updateLoanInfo, clearSummaryData } = this.props;

    return (
      <div className="loan-calculator-wrapper">
        {
          showLoanCalculator && (
            <LoanCalculator
              updateLoanInfo={updateLoanInfo}
              clearSummaryData={clearSummaryData}
            />
          )
        }
        {
          showLoanSummary && (
            <LoanCalculatorSummary
              loanInfo={loanInfo}
              summaryData={summaryData}
            />
          )
        }

      </div>
    )
  }
}

const mapStateToProps = ({ loanCalculator }) => {
  return ({
    loanCalculator
  })
}

LoanCalculatorContainer.propTypes = {
  loanCalculator: PropTypes.object.isRequired,
  updateLoanInfo: PropTypes.func.isRequired,
  clearSummaryData: PropTypes.func.isRequired,
};

LoanCalculatorContainer.defaultProps = {
  loanCalculator: {
    loanInfo: [],
    showLoanCalculator: true,
    showLoanSummary: false,
    summaryData: {}
  }
};


export default connect(mapStateToProps, mapDispatchToProps)(LoanCalculatorContainer);
