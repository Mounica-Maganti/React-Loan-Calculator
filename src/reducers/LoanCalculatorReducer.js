import * as Actions from '../actions/Constants';

const initialState = {
    showLoanCalculator: true,
    showLoanSummary: false,
    loanInfo: [],
    summaryData: {}
};

export const loanCalculatorReducer = (state = initialState, action) => {

    switch (action.type) {
        case Actions.UPDATE_LOAN_INFO:
            return {
                ...state, loanInfo: action.loanCalculatedInfo,
                summaryData: action.summaryData,
                showLoanSummary: true
            };
        case Actions.CLEAR_SUMMARY_DATA:
            return { ...state, loanInfo: action.data, showLoanSummary: action.showLoanSummary }
        default: return state;
    }
}