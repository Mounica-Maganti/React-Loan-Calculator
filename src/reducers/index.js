import { combineReducers } from 'redux';
import { loanCalculatorReducer } from './LoanCalculatorReducer';

export default combineReducers({
    loanCalculator: loanCalculatorReducer
});