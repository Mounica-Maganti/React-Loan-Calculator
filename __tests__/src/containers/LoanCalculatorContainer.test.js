import React from 'react';
import { shallow } from 'enzyme';
import { createStore } from 'redux';
import LoanCalculatorContainer from '../../../src/containers/LoanCalculatorContainer';
import reducers from '../../../src/reducers/index';


describe('Loan Calculator Container Tests', () => {
    let props;
    beforeEach(() => {
        props = {
            loanCalculator: {
                loanInfo: [],
                showLoanCalculator: true,
                showLoanSummary: false,
                updateLoanInfo: jest.fn(),
                clearSummaryData: jest.fn()
            }
        }
    });

    it('should render Loan Calculator Component', () => {
        const renderedModule = shallow(<LoanCalculatorContainer {...props} store={createStore(reducers)} />);
        expect(renderedModule).toMatchSnapshot();
    })

    it('should render Loan Summary Component', () => {
        props.showLoanSummary = true;
        const renderedModule = shallow(<LoanCalculatorContainer {...props} store={createStore(reducers)} />);
        expect(renderedModule).toMatchSnapshot();
    })
});