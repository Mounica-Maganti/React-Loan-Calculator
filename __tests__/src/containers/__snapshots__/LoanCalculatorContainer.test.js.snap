// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loan Calculator Container Tests should render Loan Calculator Component 1`] = `ShallowWrapper {}`;

exports[`Loan Calculator Container Tests should render Loan Summary Component 1`] = `ShallowWrapper {}`;
